core/src/main/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerAudioTrackSelector.java

```diff
@@ -10,19 +10,37 @@
 class AndroidMediaPlayerAudioTrackSelector {
 
     private static final int NO_FORMAT = 0;
+    private static final int NO_CHANNELS = -1;
+    private static final int NO_FREQUENCY = -1;
+    private static final String NO_MIME_TYPE = "";
+
+    private final TrackInfosFactory trackInfosFactory;
+
+    AndroidMediaPlayerAudioTrackSelector(TrackInfosFactory trackInfosFactory) {
+        this.trackInfosFactory = trackInfosFactory;
+    }
 
     List<PlayerAudioTrack> getAudioTracks(MediaPlayer mediaPlayer) {
         if (mediaPlayer == null) {
             throw new NullPointerException("You can only call getAudioTracks() when video is prepared.");
         }
 
         List<PlayerAudioTrack> audioTracks = new ArrayList<>();
-        MediaPlayer.TrackInfo[] trackInfos = mediaPlayer.getTrackInfo();
-
-        for (int i = 0; i < trackInfos.length; i++) {
-            MediaPlayer.TrackInfo trackInfo = trackInfos[i];
-            if (trackInfo.getTrackType() == MediaPlayer.TrackInfo.MEDIA_TRACK_TYPE_AUDIO) {
-                audioTracks.add(new PlayerAudioTrack(i, NO_FORMAT, String.valueOf(trackInfo.hashCode()), trackInfo.getLanguage(), "", -1, -1));
+        NoPlayerTrackInfos trackInfos = trackInfosFactory.createFrom(mediaPlayer);
```

Comment:
Couldn't mock or create a `MediaPlayer.TrackInfo` from `mediaPlayer.getTrackInfo` so we now have a `Factory` that creates our mockable `NoPlayerTrackInfos` collection that consists of `NoPlayerTrackInfo` the wrapper around `MediaPlayer.TrackInfo`.